doc/pcie-mon/pcie-monitoring-services-hld.md

```diff
@@ -0,0 +1,122 @@
+
+1. `pcieutil` should get the platform specific PCIe device information and monitor the PCIe device and bus status with PcieUtil.get_pcie_check.
+
+2. `PcieUtil` will provide APIs `load_config_file`, `get_pcie_device` and `get_pcie_check` to get the expected PCIe device list and informations, to get the current PCIe device information, and check if any PCIe device is missing or if there is any PCIe bus error.
+
+![pcieinfo_design](https://github.com/Azure/SONiC/blob/master/doc/pcieinfo_design.md)
+
+### 1.2 PCIe device configuration file ###
+
+PcieUtil needs to get the expected PCIe device information to check the PCIe device status periodically, which is different for each platform/hardware sku.
+
+Each vendor need to generate the PCIe device configuration file name as pcie.yml and locate the file under device/<platform>/<hardware_skus>/plugins. 
+
+Example) Location: `device/celestica/x86_64-cel_seastone-r0/plugins/pcie.yaml`
+
+```
+...
+- bus: '01'
+  dev: '00'
+  fn: '0'
+  id: b960
+  name: 'Ethernet controller: Broadcom Limited Broadcom BCM56960 Switch ASIC'
+- bus: '01'
+  dev: '00'
+  fn: '1'
+  id: b960
+  name: 'Ethernet controller: Broadcom Limited Broadcom BCM56960 Switch ASIC'
+```
+
+### 1.3 PCIe device status check ###
+
+
+The default PCIe device check function, get_pcie_check is implemented in PcieUtil class at sonic_platform_base/sonic_pcie/pcie_common.py.
+It loads the PCIe device configuration file and compares them with the enumerated devices based on the platform sysfs device tree under /sys/bus/pci/devices/.
+
+Here we define a common platform API to in class `PcieBase`: 
+
+    @abc.abstractmethod
+    def get_pcie_check(self, timeout=0):
+        """
+         Check Pcie device with config file
+         Returns:
+            A list including pcie device and test result info
+        """
+        return []
+
+Each vendor need to implement this function in `PcieBase` plugin if vendor has any additional pcie healthy check method.
+
+PcieUtil calls this API to check the PCIe device status, following example code showing how this API will be called:
+
+    while True:
+        status, device_dict = platform_pcieutil.get_pcie_check()
+        if(status):
+            for key, value in device_dict.iteritems():
+                print("Device on PCIe bus: %s" was %s" % (key, value))
+                 
+### 1.4 PCIe Check Service `pcie-check.service` flow ###
+
+pcie-check.service will be started by systemd during boot up and it will spawn a thread to check PCIe device status and perform the rescan pci devices if there is any missing devices after rc.local.service is completed and it will update the state db with pcie device satus after the `pcieutil pcie-chek` call so that the dependent services/container or kernel driver can be started or stopped based on the status.
+
+Detailed flow as showed in below chart: 
+![](https://github.com/Azure/SONiC/blob/master/images/pcie-check.svg)
```

Comment:
Unable to view this file.

Reply:
https://github.com/Azure/SONiC/blob/70a152f1b98e145c9f0771e7cda7a951d98a978e/images/pcie-check.svg --- This link needs to be updated